test/pkg.jl

```diff
@@ -100,6 +100,8 @@ temp_pkg_dir() do
     str = chomp(takebuf_string(iob))
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
+    # Checking to see dependency reading
+    @test Pkg.dependencies("Example") == Dict("julia" => Pkg.Types.VersionSet(v"0.3.0", typemax(VersionNumber)))
```

Comment:
If there is a breaking change in how packages are structured or what information they contain, wouldn't it be important for this test to use an up-to-date package?

Ideally I could mock the file with something that's supposed to generate a "spec" `REQUIRE` file like PkgDev, but I feel like that wouldn't quite be kosher/possible as it's out of Base now.